src/processing/getFiatRate/getDayAvgCurrencyRate/index.ts

```diff
@@ -19,8 +19,27 @@ export async function getDayAvg(
             avgType: type,
         }
     });
-    const rate = cryptocompareRateResponse(response, fiatCurrency);
-    return rate || 0;
+    let rate = 0;
+    if ('data' in response) {
+        try {
+            const result = response.data;
+            if ('RateLimit' in result && 'max_calls' in result.RateLimit) {
+                    await waitForRateLimit(result.RateLimit);
+                    getDayAvg(fiatCurrency, currency, date, type).then(
```

Comment:
Cant you just use await here and wrap in another try/catch?

Reply:
Yes